### ISSUE TITLE
Missing npcap loopback interface in Wireshark

### ISSUE THREAD

**Issue by chauser** [nmap/nmap#802]
TL;DR: A few people have reported that after installing npcap the loopback interface is not listed in Wireshark even though it shows up, e.g., in the ipconfig list. In my case the reason turned out to be that the Windows Basic Filtering Enging (BFE) service was not running. After starting the BFE service, Wireshark listed the npcap loopback interface as an available capture interface and loopback packet capture worked correctly.

Long explanation: If the BFE is disabled and you don't know that it could be an issue it takes a very long time to discover that it is the source of the problem -- there is only the barest hint in the log output of the debug version of packet.dll of what is wrong and it can only be recognized as such by exploration in the registry, by building the debug version of packet.dll, etc. 

Suggestion: document in the npcap README that the BFE service must be running and that the loopback capture interface will not show up in Wireshark if it isn't. It might be possible to have the npcap driver log an error message if it unable to open  loopback, or have packet.dll log an error if it is unable to open the loopback interface, along with the error code. But few enough people seem to experience this that simply documenting the behavior is probably sufficient to eliminate almost all the confusion that it causes.

**Comment by Varunram** [comment]
Thanks Carl for your expert analysis of this frequently faced problem, both here and at http://seclists.org/nmap-dev/2017/q1/239. We should look into how we can make users aware of this (maybe a "Frequent Bugs" section in the README?)

Maybe Related - #628

**Comment by hsluoyz** [comment]
Maybe [Q&A in the docs](https://rawgit.com/nmap/npcap/master/docs/npcap-guide-wrapper.html#npcap-qa) is a good location to record this.

**Comment by hsluoyz** [comment]
I confirmed this issue on Npcap 0.90, Wireshark 2.2.2 on Win10 x64.

**Comment by hsluoyz** [comment]
@chauser, the advice has been added here: https://rawgit.com/nmap/npcap/master/docs/npcap-guide-wrapper.html#npcap-qa.

Thanks for the report!

**Comment by chauser** [comment]
Thanks, Yang. That should take care of it for anyone else who has that
problem. -- Carl

**Comment by Asday** [comment]
Hasn't fixed it for me; BFE was already started.  Trying to restart it just gave "Access Denied".

I can see the adapter in Network Connections, but Wireshark refuses to see it, and all the documentation surrounding this leaves somewhat to be desired.

**Comment by VioletGiraffe** [comment]
@Asday Same for me. Did you ever figure out the solution? Win 10 Pro x64 here.

**Comment by niklasholm** [comment]
Same here, BFE already started, loopback device visible in network devices and ipconfig, but not in Wireshark :( @hsluoyz, any advice?

**Comment by hsluoyz** [comment]
@niklasholm Wireshark calls into ``wpcap.dll``, which calls into ``packet.dll`` to get the loopback device. If it's not there, I guess the ``OpenAdapter()`` function in ``packet.dll`` fails for loopback device. You can compile ``packet.dll``'s source code and debug that function to troubleshoot why it fails.
